tests/Dockerfile

```diff
@@ -0,0 +1,36 @@
+FROM rust:latest
+
+# Install Node.js
+ENV NODE_VERSION 12.0.0
+RUN curl https://raw.githubusercontent.com/creationix/nvm/master/install.sh | bash \
+  && . $HOME/.nvm/nvm.sh \
+  && nvm install $NODE_VERSION \
+  && nvm alias default $NODE_VERSION \
+  && nvm use default
+ENV PATH /root/.nvm/versions/node/v$NODE_VERSION/bin:$PATH
+# Check that Node.js was correctly installed
+RUN node --version 
+
+# Create blank project
+RUN USER=root cargo new --bin starship
+WORKDIR /starship
+
+# We want dependencies cached, so copy those first
+COPY ./Cargo.lock ./Cargo.lock 
+COPY ./Cargo.toml ./Cargo.toml 
```

Comment:
`The docker image does not rebuild if dependencies change` 

Each command here will run and create a layer / image. Everytime the docker image rebuild it will check the hash and either use the cache or re run the command. So in this case it will hash the contents of the `Cargo.toml` and use that as the hash. So if the contents change it will rerun this command and all below it.

Reply:
Correct. I have intentionally made the `integration_test` script skip `docker build` if the image already exists because of the time required to send the build context to the docker daemon when `target/` is built. It's usually around 1GB, so it can take a while.

It looks like adding `.dockerignore` to the project root with `target/` solves it. 😄